client/js/shout.js

```diff
@@ -682,18 +658,47 @@ $(function() {
 			"quit",
 			"nick",
 			"mode",
+			"whois"
 		];
 		if ($.inArray(type, ignore) !== -1){
 			return;
 		}
 
-		var badge = button.find(".badge");
-		if (badge.length !== 0) {
-			var i = (badge.data("count") || 0) + 1;
-			badge.data("count", i);
-			badge.html(i > 999 ? (i / 1000).toFixed(1) + "k" : i);
-			if (highlight || isQuery) {
-				badge.addClass("highlight");
+		var isQuery = button.hasClass("query");
+		var highlight = type.contains("highlight");
+		var message = type.contains("message");
+		var settings = $.cookie("settings") || {};
+		if (highlight || isQuery || (settings.notifyAllMessages && message)) {
+			if (settings.notification) {
+				pop.play();
+			}
+			favico.badge("!");
+			if (settings.badge && Notification.permission === "granted") {
+				var notify = new Notification(msg.from + " says:", {
+					body: msg.text.trim(),
+					icon: "/img/logo-64.png",
+					tag: target
+				});
+				notify.onclick = function() {
+					window.focus();
+					button.click();
+					this.close();
+				};
+				window.setTimeout(function() {
+					notify.close();
+				}, 5 * 1000);
+			}
+		}
+
+		if (!button.hasClass("active")) {
```

Comment:
Ah yeah, that makes sense. Just cache `hasClass` result into a variable, so you're not calling it twice.